### ISSUE TITLE
Investment Intelligence Platform: Models, Scoring Engine & UI

### ISSUE THREAD

**Issue by holden** [razrfly/pokevestment#26]
## Vision

A user-facing interface that answers: **"What should I buy today?"** — with drill-down by series, set, rarity, artist, character, and custom criteria. Backed by ML models that find market misvaluations where the current price is below what historical patterns predict the card should be worth.

This issue covers the three missing layers: **scoring models**, **data pipeline for model outputs**, and **the UI to visualize it all**.

---

### What We Have (sufficient for Model 1)
- 22,754 cards with 40+ features each
- 19,039 cards with pricing from TCGPlayer + CardMarket (83.7% coverage)
- 1,025 Pokemon species with legendary/mythical/baby flags, capture rate, evolution chains
- 407 distinct illustrators
- 200 sets with release dates across 21 series
- Daily price sync accumulating via Oban cron

### What We Lack (needed for advanced models)
| Gap | Impact | When Available | Tracked In |
|---|---|---|---|
| Time-series depth (need 30+ days) | Blocks trend/trajectory models | ~30 days (accumulating now) | Automatic |
| Tournament meta data | Highest-signal short-term predictor | When #25 is implemented | #25 |
| Historical price backfill | Accelerates trajectory models | Research phase | #15 |
| Art type extraction | Medium-high signal, easy win | When #25 P2 is done | #25 |
| Set supply proxies (era, secret ratio) | Medium signal | When #25 P3 is done | #25 |

### Verdict
**Enough to build and ship Model 1 + basic UI now.** Advanced models unlock progressively as data accumulates.

---

### Model 1: Fair Value Predictor (BUILD NOW)

**Question**: Given a card's attributes, what should its market price be?

**Approach**: Gradient boosted trees (XGBoost/LightGBM) trained on 19K+ priced cards.

**Input features** (available today):
- Card: rarity, category, suffix, stage, hp, attack_count, max_attack_damage, has_ability, ability_count, weakness_count, resistance_count, retreat_cost, is_secret_rare, first_edition, is_shadowless
- Species: is_legendary, is_mythical, is_baby, capture_rate, generation, color
- Set: release_date (→ age_days), card_count_official, series_id, regulation_mark
- Illustrator: target-encoded or frequency-encoded
- Type: primary energy type, type count

**Output**: Predicted fair market value (USD)

**Key derived metric — Value Gap Score**:
```
value_gap = (predicted_fair_value - current_market_price) / current_market_price
```
- `value_gap > 0.5` → Card is >50% undervalued → **Strong Buy signal**
- `value_gap > 0.25` → Moderately undervalued → **Buy signal**
- `value_gap ≈ 0` → Fairly priced
- `value_gap < -0.25` → Overpriced → **Avoid/Sell signal**

**Confidence scoring**: Model prediction interval width. Narrow interval = high confidence.

### Model 2: Illustrator Premium Model (BUILD NOW)

**Question**: Does this artist independently add value after controlling for other factors?

407 illustrators — some command consistent premiums. This is genuine alpha most collectors miss.

**Output**: Per-illustrator premium multiplier (e.g., "Mitsuhiro Arita cards trade at 1.8x predicted value on average").

### Model 3: Price Trajectory Predictor (BUILD AT 60+ DAYS)

**Question**: Which direction is this card's price heading?

**Features**: 7d/30d moving averages, volatility, price momentum, CardMarket avg1/avg7/avg30 spreads.

**Output**: 30-day price direction (up/down/stable) with magnitude estimate.

### Model 4: Meta Impact Scorer (BUILD AFTER #25)

**Question**: How does competitive play affect this card's value?

**Features**: meta_share, tournament_appearances, avg_win_rate, archetype_count, meta_trend.

**Output**: "Meta premium" — how much competitive demand adds to fair value.

### Combined Investment Score

```
investment_score = value_gap × confidence × trajectory_multiplier × (1 + meta_premium)
```

Cards ranked by this composite score = **"Best Buys Right Now"**.

---

## Phase 2: Model Output Storage

Models need to write predictions somewhere the UI can read them. These change frequently (daily or weekly retraining).

### New Table: `card_valuations`

Stores the latest model predictions per card. Overwritten on each model run.

| Column | Type | Notes |
|---|---|---|
| `id` | serial PK | |
| `card_id` | string, FK → cards, unique | One active valuation per card |
| `predicted_value` | decimal(10,2) | Model 1 fair value estimate (USD) |
| `prediction_low` | decimal(10,2) | Lower bound of prediction interval |
| `prediction_high` | decimal(10,2) | Upper bound of prediction interval |
| `value_gap` | decimal(8,4) | (predicted - market) / market |
| `confidence` | decimal(4,3) | 0.000 - 1.000 |
| `signal` | string | "strong_buy", "buy", "fair", "overpriced", "strong_sell" |
| `investment_score` | decimal(8,4) | Composite ranking score |
| `trajectory` | string, nullable | "rising", "falling", "stable" (null until Model 3) |
| `trajectory_magnitude` | decimal(6,4), nullable | Predicted % change in 30d |
| `meta_premium` | decimal(6,4), nullable | Tournament demand premium (null until Model 4) |
| `illustrator_premium` | decimal(6,4) | Artist premium multiplier from Model 2 |
| `top_features` | jsonb | Top 3-5 features driving this card's valuation |
| `model_version` | string | e.g. "v1.0.0" for tracking |
| `scored_at` | utc_datetime | When this score was computed |
| `timestamps` | utc_datetime | |

**Indexes**: unique on `card_id`, btree on `signal`, btree on `investment_score DESC`, btree on `value_gap DESC`

### New Table: `valuation_snapshots`

Historical record of how valuations change over time. Append-only.

| Column | Type | Notes |
|---|---|---|
| `id` | serial PK | |
| `card_id` | string, FK → cards | |
| `predicted_value` | decimal(10,2) | |
| `value_gap` | decimal(8,4) | |
| `investment_score` | decimal(8,4) | |
| `signal` | string | |
| `model_version` | string | |
| `snapshot_date` | date | |
| `timestamps` | utc_datetime | |

**Indexes**: `(card_id, snapshot_date)` unique, btree on `snapshot_date`

This lets us track: "Has this card been consistently undervalued for 2 weeks?" (persistent opportunity) vs "This just became undervalued today" (new opportunity).

### New Table: `model_runs`

Metadata about each model training/scoring run.

| Column | Type | Notes |
|---|---|---|
| `id` | serial PK | |
| `model_type` | string | "fair_value", "illustrator", "trajectory", "meta" |
| `model_version` | string | Semver |
| `cards_scored` | integer | How many cards were scored |
| `training_samples` | integer | Training set size |
| `accuracy_metrics` | jsonb | R², MAE, MAPE, feature importances |
| `ran_at` | utc_datetime | |
| `duration_seconds` | integer | |
| `timestamps` | utc_datetime | |

---

### 3A: Dashboard — "Best Buys Right Now"

The landing page. Shows the highest-conviction investment opportunities across the entire catalog.

```
┌─────────────────────────────────────────────────────────────┐
│  POKEVESTMENT — Investment Intelligence Dashboard           │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  📊 Market Summary (header bar)                             │
│  ┌──────────┐ ┌──────────┐ ┌──────────┐ ┌──────────────┐   │
│  │ 247      │ │ 1,892    │ │ 412      │ │ Model v1.2   │   │
│  │ Strong   │ │ Buy      │ │ Over-    │ │ Last scored: │   │
│  │ Buys     │ │ Signals  │ │ priced   │ │ 2h ago       │   │
│  └──────────┘ └──────────┘ └──────────┘ └──────────────┘   │
│                                                             │
│  🔥 Top Opportunities (sorted by investment_score DESC)     │
│  ┌─────────────────────────────────────────────────────┐    │
│  │ [Card Image] Pikachu VMAX (Secret)                  │    │
│  │ Set: Vivid Voltage | Artist: HYOGONOSUKE            │    │
│  │ Market: $45.00 | Fair Value: $89.00 | Gap: +97%     │    │
│  │ Signal: STRONG BUY | Confidence: 94%                │    │
│  │ Why: Secret rare + iconic Pokemon + top illustrator  │    │
│  │      premium. Historically Gen 1 secrets in modern   │    │
│  │      sets converge to $80-120 range.                 │    │
│  ├─────────────────────────────────────────────────────┤    │
│  │ [Card Image] Umbreon VMAX (Alt Art)                 │    │
│  │ ...                                                  │    │
│  └─────────────────────────────────────────────────────┘    │
│                                                             │
│  Filters: [Series ▾] [Set ▾] [Min Price ▾] [Signal ▾]     │
│           [Rarity ▾] [Generation ▾] [Type ▾]               │
│           [Artist ▾] [Pokemon ▾] [Sort By ▾]               │
│                                                             │
└─────────────────────────────────────────────────────────────┘
```

**Key features**:
- Card image thumbnail + key stats at a glance
- "Why" explanation powered by `top_features` from the model
- Filters that compose (AND logic): narrow by series, rarity, price range, generation, etc.
- Sort by: investment_score, value_gap, confidence, price
- Pagination with infinite scroll or paginated results

### 3B: Series/Set Drill-Down

```
┌─────────────────────────────────────────────────────────────┐
│  Scarlet & Violet Series                                    │
│  14 sets | 4,200 cards | 312 buy signals                   │
├─────────────────────────────────────────────────────────────┤
│                                                             │
│  Set Overview (sorted by opportunity density)               │
│  ┌────────────────────────────────────────────────────┐     │
│  │ Prismatic Evolutions (sv08.5)                      │     │
│  │ Released: 2025-01-17 | 233 cards                   │     │
│  │ 🟢 42 Strong Buys | 🟡 67 Buys | 🔴 18 Overpriced │     │
│  │ Avg Value Gap: +34% | Set Total Value: $12,450     │     │
│  │ [View Set →]                                       │     │
│  ├────────────────────────────────────────────────────┤     │
│  │ Surging Sparks (sv08) ...                          │     │
│  └────────────────────────────────────────────────────┘     │
│                                                             │
│  Click into a set → card-level view with all signals        │
│                                                             │
└─────────────────────────────────────────────────────────────┘
```

### 3C: Card Detail Page

```
┌─────────────────────────────────────────────────────────────┐
│  Pikachu VMAX (Secret) — Vivid Voltage #188                 │
├──────────────────────┬──────────────────────────────────────┤
│                      │  INVESTMENT ANALYSIS                  │
│  [Large Card Image]  │                                       │
│                      │  Signal: 🟢 STRONG BUY                │
│                      │  Investment Score: 8.7 / 10            │
│                      │  Confidence: 94%                       │
│                      │                                       │
│                      │  Market Price: $45.00 (TCGPlayer)     │
│                      │  Fair Value: $89.00 (Model v1.2)      │
│                      │  Value Gap: +97.8%                    │
│                      │                                       │
│                      │  ── Value Drivers ──                  │
│                      │  1. Secret rare (65x common premium)  │
│                      │  2. Generation 1 Pokemon (+25x)       │
│                      │  3. HYOGONOSUKE illustrator (+2.1x)   │
│                      │  4. VMAX suffix (+3.4x)               │
│                      │  5. Set age: 1,800 days (vintage      │
│                      │     premium building)                  │
│                      │                                       │
├──────────────────────┴──────────────────────────────────────┤
│  PRICE HISTORY                                               │
│  [Line chart: market price vs predicted fair value over      │
│   time, showing convergence/divergence]                      │
│                                                              │
│  SIMILAR CARDS (comparable investment profiles)              │
│  [Grid of 4-6 cards with similar features + their signals]  │
│                                                              │
│  VALUATION HISTORY                                           │
│  [Timeline: how the model's score for this card has changed] │
└──────────────────────────────────────────────────────────────┘
```

### 3D: Insights Views

**By Artist**: Which illustrators produce the most undervalued cards?
- Table: illustrator name, card count, avg premium, cards currently undervalued
- "Mitsuhiro Arita has 145 cards, 23 currently undervalued, avg gap +41%"

**By Pokemon/Character**: Which Pokemon are undervalued across all their printings?
- Group by species, show avg value gap across all cards featuring that Pokemon
- "Gengar: 47 cards, 12 undervalued, strongest buy: Gengar ex SIR from Ascended Heroes"

**By Generation**: Which era of Pokemon is the best value right now?
- Gen 1 nostalgia vs Gen 9 competitive demand

**Feature Importance Report**: What actually drives card value?
- Bar chart of feature importances from the model
- "Rarity explains 28% of price variation, followed by generation (15%), illustrator (12%)..."

---

### ML Pipeline

**Decision: Python sidecar vs Elixir/Nx**

Option A (Recommended): **Python sidecar** — Train models in Python (XGBoost/scikit-learn), write predictions directly to PostgreSQL, Phoenix app reads predictions.
- Pro: Mature ML ecosystem, XGBoost/LightGBM, pandas, scikit-learn
- Pro: Can run as a scheduled script (cron/Oban triggers a mix task that calls Python)
- Con: Two languages in the stack

Option B: **Elixir/Nx** — Stay in Elixir ecosystem with Nx/Scholar
- Pro: Single language
- Con: ML ecosystem is immature, no XGBoost equivalent, limited feature engineering tools

**Recommendation**: Option A. The ML pipeline is a batch job that runs daily/weekly. It doesn't need to be in the web framework. Python → PostgreSQL → Phoenix reads the results.

### Retraining Schedule
- **Model 1 (Fair Value)**: Retrain weekly as new price data arrives
- **Model 2 (Illustrator)**: Retrain monthly
- **Model 3 (Trajectory)**: Retrain weekly once 60+ days of data exist
- **Model 4 (Meta)**: Retrain after each major tournament

### UI Stack
Phoenix LiveView — real-time updates, server-rendered, already in the stack. No separate frontend needed.

**Components needed**:
- Card list with filters (LiveView component with phx-change on filter controls)
- Card detail page (standard controller + template)
- Charts (price history, feature importance) — [VegaLite](https://github.com/livebook-dev/vega_lite) or a JS charting library
- Responsive card grid with image thumbnails

---

### Phase 1: Schema + Model 1 (Week 1-2)
- [ ] Create `card_valuations`, `valuation_snapshots`, `model_runs` tables (migration)
- [ ] Complete feature engineering easy wins (#19)
- [ ] Build Python ML pipeline: feature matrix assembly → XGBoost training → PostgreSQL write
- [ ] Train Model 1 on current data, generate first predictions
- [ ] Generate feature importance report
- [ ] First "undervalued cards" list

### Phase 2: Basic UI (Week 2-3)
- [ ] Dashboard page: top opportunities list with card images, signals, value gaps
- [ ] Filter system: series, set, rarity, generation, type, price range, signal
- [ ] Card detail page: investment analysis, value drivers, card metadata
- [ ] Series/set drill-down view
- [ ] Basic sorting (investment score, value gap, confidence, price)

### Phase 3: Enriched Views (Week 3-4)
- [ ] Artist insights page (illustrator premium analysis)
- [ ] Pokemon/character insights page
- [ ] Generation comparison view
- [ ] Feature importance visualization
- [ ] Price history chart (once 30+ days accumulated)

### Phase 4: Advanced Models (Week 8+, after 60 days of price data)
- [ ] Model 3: Price trajectory predictions
- [ ] Model 4: Meta impact scoring (after #25 tournament data)
- [ ] Combined investment score with all model inputs
- [ ] "New Set Drop" report — automatic analysis when new sets appear
- [ ] Valuation change alerts — cards that just became undervalued

---

## Dependencies

```
#19 (Feature Engineering)  ─┐
#25 (Schema Normalization) ─┤
                            ├──→ THIS ISSUE (Phase 1-4)
#12 (ML Strategy)          ─┤
#14 (New Set Advisor)      ─┘
                              ↓
                         #15 (Historical Backfill) — accelerates Phase 4
                         #20 (Data Gaps) — improves model quality
```

**Hard dependencies for Phase 1-2**: None — can start now with existing data.
**Soft dependencies**: #19 and #25 improve model quality but aren't blockers.
**Phase 4 blockers**: 60+ days of accumulated price snapshots, #25 for tournament data.

## Success Criteria

- [ ] Model 1 trained with R² ≥ 0.6 and MAPE ≤ 50% on held-out test set
- [ ] Feature importance report identifies non-obvious value drivers
- [ ] Dashboard shows ranked buy opportunities with explanations
- [ ] User can filter/drill-down by series, set, rarity, generation, type, artist, Pokemon
- [ ] Card detail page shows investment thesis with top value drivers
- [ ] Predictions refresh automatically (weekly minimum)
- [ ] At least 3 "undervalued" cards identified that appreciate 25%+ within 60 days (validation)

## Labels
feature, ml, ui, investment-intelligence

## Related Issues
- #12 (ML Strategy — model specifications)
- #14 (New Set Advisor — roadmap context)
- #19 (Feature Engineering — model input improvements)
- #20 (Data Gaps — future signal sources)
- #25 (Schema Normalization — tournament data + art type extraction)
